stm_workshop/lib/data/data.hpp

```diff
@@ -0,0 +1,130 @@
+#pragma once
+
+#include <array>
+#include <cstdint>
+#include <optional>
+#include <unordered_map>
+#include <vector>
+
+namespace hyped::data {
+
+// -------------------------------------------------------------------------------------------------
+// Global Module States
+// -------------------------------------------------------------------------------------------------
+enum class ModuleStatus {
+  kCriticalFailure,  // state machine transitions to EmergencyBraking/FailureStopped
+  kStart,            // Initial module state
+  kInit,   // state machine transistions to Calibrating if all modules have Init or Ready status.
+  kReady,  // state machine transistions to Ready if all modules have Ready status.
+};
+
+struct Module {
+  ModuleStatus module_status = ModuleStatus::kStart;
+};
+
+// -------------------------------------------------------------------------------------------------
+// Navigation
+// -------------------------------------------------------------------------------------------------
+using nav_t            = float;
+struct Navigation : public Module {
+  static constexpr nav_t kMaximumVelocity    = 100;        // m/s
+  static constexpr nav_t kRunLength          = 1250.0;     // m
+  static constexpr nav_t kBrakingBuffer      = 20.0;       // m
+  nav_t displacement;                                      // m
+  nav_t velocity;                                          // m/s
```

Comment:
`nav_t acceleration;` as well?

Reply:
Since we don't have any transitions requiring it, I thought to just omit it to avoid confusion